### PR TITLE
feat: add clickbench benchmark

### PR DESCRIPTION
**Pull request by AdamGS**
Adds a [clickbench](https://github.com/ClickHouse/ClickBench) run to our datafusion benchmarks. ~Some queries fail, see #1326.~
Running this is pretty resource intensive as the dataset is ~100m rows/15GB of parquet.

### CODE REVIEW

File: bench-vortex/src/tpch/execute.rs [hunk 1]
```diff
@@ -44,9 +43,11 @@ pub async fn run_tpch_query(
 
 pub async fn execute_query(ctx: &SessionContext, query: &str) -> Result<Vec<RecordBatch>> {
     let plan = ctx.sql(query).await?;
-    let (state, plan) = plan.into_parts();
-    let optimized = state.optimize(&plan)?;
-    let physical_plan = state.create_physical_plan(&optimized).await?;
-    let result = collect(physical_plan.clone(), state.task_ctx()).await?;
-    Ok(result)
+    plan.collect().await
```
Comment (robert3005) [inline bench-vortex/src/tpch/execute.rs]: I thought this doesn't run all the optimizations steps

Comment (AdamGS) [reply bench-vortex/src/tpch/execute.rs]: yeah, commented it out when I was debugging some issue, I'll add it back in a second

**Comment by AdamGS** [comment]
some benchmarks are failing, I'll fix them tomorrow

File: bench-vortex/src/clickbench.rs [hunk 2]
```diff
@@ -0,0 +1,264 @@
...
+    session: &SessionContext,
+    table_name: &str,
+    file: &Path,
+    schema: &Schema,
+    enable_compression: bool,
+) -> anyhow::Result<()> {
+    let vortex_dir = file.parent().unwrap().join(if enable_compression {
+        "vortex_compressed"
+    } else {
+        "vortex_uncompressed"
```
Comment (lwwmanning) [inline bench-vortex/src/clickbench.rs]: No real point in uncompressed file?

Comment (AdamGS) [reply bench-vortex/src/clickbench.rs]: definitely some more cleanup to do here, ironing out some bugs first.

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -150,7 +150,7 @@ jobs:
 
   bench-test:
     name: "bench test"
-    runs-on: ubuntu-latest
```
Comment (lwwmanning) [inline .github/workflows/ci.yml]: does this make bench test materially faster?

File: bench-vortex/benches/clickbench.rs [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+
+    let raw_data = download_data(
+        basepath.join("hits.parquet"),
+        "https://datasets.clickhouse.com/hits_compatible/hits.parquet",
+    );
+
+    let output_path = basepath.join("processed.parquet");
+
+    let final_parquet_path = idempotent(&output_path, |output_path| {
+        println!("Fixing parquet file");
```
Comment (lwwmanning) [inline bench-vortex/benches/clickbench.rs]: lol wat...? what's the story here?

Comment (AdamGS) [reply bench-vortex/benches/clickbench.rs]: The parquet file clickbench provides misses some more "complex" types, so we need to either cast everything on query time, or re-process it to have the correct timestamp columns. See clickhouse/clickbench#7.

File: .github/workflows/ci.yml (hunk 3 above)
Comment (AdamGS) [reply .github/workflows/ci.yml]: actually not necessary, I'll revert it.

File: bench-vortex/benches/clickbench.rs (hunk 4 above)
Comment (lwwmanning) [reply bench-vortex/benches/clickbench.rs]: cool. would be good to put a comment with a link to that issue

**Review by lwwmanning — APPROVED** [review]

**Comment by AdamGS** [comment]
Running clickbench OOMs our dedicated runners, I'll re-generate the file locally tomorrow and put it in some accessible location for future runs.

**Comment by gatesn** [comment]
@AdamGS can we make this a bin/clickbench like we did for TPCH?

**Comment by danking** [comment]
@AdamGS fwiw, I put the two parquet files in s3://vortex-bench-dev/clickbench/

**Comment by danking** [comment]
🚢

**Comment by AdamGS** [comment]
I want to pull the data from AWS first, and then I'll merge

**Pull request merged**